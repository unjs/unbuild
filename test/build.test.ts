import { describe, it, expect } from "vitest";
import fs from "node:fs";
import path from "node:path";
import { fileURLToPath } from "node:url";
import { build } from "../src/index.js";

type BuildConfig = Parameters<typeof build>[2];

describe("Build fixtures", () => {
  const tests: [string, BuildConfig & { dir: string }][] = [
    [
      "declaration (mkdist)",
      {
        dir: "fixtures/mkdist-declaration",
      },
    ],
    [
      "sourcemap (mkdist)",
      {
        dir: "fixtures/mkdist-sourcemap",
      },
    ],
    [
      "declaration (rollup)",
      {
        dir: "fixtures/rollup-declaration",
      },
    ],
    [
      "sourcemap (rollup)",
      {
        dir: "fixtures/rollup-sourcemap",
      },
    ],
  ];

  it.each(tests)("%s", async (_, { dir, ...buildOptions }) => {
    const cwd = new URL(dir.replace(/\/?$/, "/"), import.meta.url);
    await build(fileURLToPath(cwd), false, buildOptions);
    for (const file of fs.readdirSync(new URL("dist/", cwd), {
      recursive: true,
      withFileTypes: true,
    })) {
      if (file.isFile()) {
        expect(
          fs.readFileSync(path.join(file.parentPath, file.name), "utf8"),
        ).toMatchSnapshot(file.name);
      }
    }
  });
});
