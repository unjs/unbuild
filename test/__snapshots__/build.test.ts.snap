// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Build fixtures > declaration (mkdist) > a.d.ts 1`] = `
"export declare function foo(): number;
"
`;

exports[`Build fixtures > declaration (mkdist) > a.mjs 1`] = `
"export function foo() {
  return 42;
}
"
`;

exports[`Build fixtures > declaration (mkdist) > index.d.ts 1`] = `
"export { foo } from "./a.js";
"
`;

exports[`Build fixtures > declaration (mkdist) > index.mjs 1`] = `
"export { foo } from "./a.js";
"
`;

exports[`Build fixtures > declaration (rollup) > index.cjs 1`] = `
"'use strict';

function foo() {
  return 42;
}

exports.foo = foo;
"
`;

exports[`Build fixtures > declaration (rollup) > index.d.cts 1`] = `
"declare function foo(): number;

export { foo };
"
`;

exports[`Build fixtures > declaration (rollup) > index.d.mts 1`] = `
"declare function foo(): number;

export { foo };
"
`;

exports[`Build fixtures > declaration (rollup) > index.d.ts 1`] = `
"declare function foo(): number;

export { foo };
"
`;

exports[`Build fixtures > declaration (rollup) > index.mjs 1`] = `
"function foo() {
  return 42;
}

export { foo };
"
`;

exports[`Build fixtures > sourcemap (mkdist) > a.mjs 1`] = `
"export function foo() {
  return 42;
}
"
`;

exports[`Build fixtures > sourcemap (mkdist) > index.mjs 1`] = `
"export { foo } from "./a.js";
"
`;

exports[`Build fixtures > sourcemap (rollup) > index.cjs 1`] = `
"'use strict';

function foo() {
  return 42;
}

exports.foo = foo;
//# sourceMappingURL=index.cjs.map
"
`;

exports[`Build fixtures > sourcemap (rollup) > index.cjs.map 1`] = `"{"version":3,"file":"index.cjs","sources":["../src/a.ts"],"sourcesContent":["export function foo(): number {\\n  return 42;\\n}\\n"],"names":[],"mappings":";;AAAO,SAAS,GAAc,GAAA;AAC5B,EAAO,OAAA,EAAA;AACT;;;;"}"`;

exports[`Build fixtures > sourcemap (rollup) > index.mjs 1`] = `
"function foo() {
  return 42;
}

export { foo };
//# sourceMappingURL=index.mjs.map
"
`;

exports[`Build fixtures > sourcemap (rollup) > index.mjs.map 1`] = `"{"version":3,"file":"index.mjs","sources":["../src/a.ts"],"sourcesContent":["export function foo(): number {\\n  return 42;\\n}\\n"],"names":[],"mappings":"AAAO,SAAS,GAAc,GAAA;AAC5B,EAAO,OAAA,EAAA;AACT;;;;"}"`;
